frontend/src/Components/SearchBar/SearchBar.jsx

```diff
@@ -1,90 +1,82 @@
-import React, { useState, useEffect, useRef } from "react";
+import React, { useState, useEffect } from "react";
+import debounce from "lodash.debounce";
 import "./SearchBar.css";
 
-const SearchBar = ({ onSelect, onSearchOnMap }) => {
+function SearchBar({ userId, onSelect }) {
   const [query, setQuery] = useState("");
-  const [results, setResults] = useState([]);
-  const [showDropdown, setShowDropdown] = useState(false);
-  const latestRequest = useRef(null);
+  const [suggestions, setSuggestions] = useState([]);
 
-  const handleInputChange = async (e) => {
-    const value = e.target.value;
-    setQuery(value);
+  const fetchSuggestions = debounce(async (input) => {
+    if (input.length > 0) {
+      try {
+        const response = await fetch(
+          `http://localhost:3000/api/search?query=${input}&userId=${userId}`
```

Comment:
putting the search query in the URL makes me nervous, but I'm not sure how this could be broken.  Could you try searching with weird characters such as `#@![?` and see if the server is still able to parse the query and userId?

Reply:
I tested the search functionality with various special characters and the server was able to parse the query and userId correctly, which suggests that the server and browser are handling the encoding automatically.